From 8f9a7703a5cb79ccc9c223c0e084b32523cd4ae1 Mon Sep 17 00:00:00 2001
From: "Dillibabu, Karthikx" <karthikx.dillibabu@intel.com>
Date: Fri, 23 Apr 2021 17:21:21 +0530
Subject: [PATCH 060/109] Fix to resolve the SPI-TPM and SPI-NOR detection
 issue

Removed the incorrect csgpiod check for the spi-mem transfer.

Signed-off-by: Dillibabu, Karthikx <karthikx.dillibabu@intel.com>
---
 drivers/spi/spi-mem.c | 2 +-
 drivers/spi/spi.c     | 8 ++------
 2 files changed, 3 insertions(+), 7 deletions(-)

diff --git a/drivers/spi/spi-mem.c b/drivers/spi/spi-mem.c
index 37f4443ce9a0..3a8200920b29 100644
--- a/drivers/spi/spi-mem.c
+++ b/drivers/spi/spi-mem.c
@@ -312,7 +312,7 @@ int spi_mem_exec_op(struct spi_mem *mem, const struct spi_mem_op *op)
 	if (!spi_mem_internal_supports_op(mem, op))
 		return -ENOTSUPP;
 
-	if (ctlr->mem_ops && !mem->spi->cs_gpiod) {
+	if (ctlr->mem_ops) {
 		ret = spi_mem_access_start(mem);
 		if (ret)
 			return ret;
diff --git a/drivers/spi/spi.c b/drivers/spi/spi.c
index 2ee79e02afea..977893d0fb08 100644
--- a/drivers/spi/spi.c
+++ b/drivers/spi/spi.c
@@ -917,12 +917,8 @@ static void spi_set_cs(struct spi_device *spi, bool enable, bool force)
 		}
 		/* Some SPI masters need both GPIO CS & slave_select */
 		if ((spi->controller->flags & SPI_MASTER_GPIO_SS) &&
-		    spi->controller->set_cs) {
-			if (spi->cs_gpiod)
-				spi->controller->set_cs(spi, enable1);
-			else
-				spi->controller->set_cs(spi, !enable);
-		}
+				spi->controller->set_cs)
+			spi->controller->set_cs(spi, !enable);
 	} else if (spi->controller->set_cs) {
 		spi->controller->set_cs(spi, !enable);
 	}
-- 
2.25.1

